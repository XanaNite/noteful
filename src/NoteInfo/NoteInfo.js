import React from 'react';
import { Link} from 'react-router-dom';
import './NoteInfo.css';

export default function NoteInfo(props){
    return(
        <div className='Note'>
            <h2 className='Note__title'>
                <Link to={`/note/${props.id}`}>
                    {props.name}
                </Link>
            </h2>
            <button className='Note__delete' type='button'>Delete Note</button>
            <div className='Note__modified_date'>{props.modified}</div>
        </div>
    );
}