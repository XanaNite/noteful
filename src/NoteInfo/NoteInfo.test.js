import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoteInfo from './NoteInfo';

describe(`NoteInfo component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Note by default', () => {
    const wrapper = shallow(<NoteInfo />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Note given props', () => {
    const wrapper = shallow(<NoteInfo {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})