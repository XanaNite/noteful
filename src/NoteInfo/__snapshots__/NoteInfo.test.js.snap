// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteInfo component renders a .Note by default 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/undefined"
    />
  </h2>
  <button
    className="Note__delete"
    type="button"
  >
    Delete Note
  </button>
  <div
    className="Note__modified_date"
  />
</div>
`;

exports[`NoteInfo component renders the Note given props 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Note__delete"
    type="button"
  >
    Delete Note
  </button>
  <div
    className="Note__modified_date"
  >
    <Component />
  </div>
</div>
`;
